internal/salesforce/client.go

```diff
@@ -0,0 +1,95 @@
+package salesforce
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/canonical/user-verification-service/internal/logging"
+	"github.com/canonical/user-verification-service/internal/monitoring"
+	"github.com/canonical/user-verification-service/internal/tracing"
+	"github.com/k-capehart/go-salesforce/v2"
+)
+
+var ErrInvalidTotalSize = fmt.Errorf("invalid total size")
+
+const query = "SELECT fHCM2__Email__c, Employment_Record_Active__c FROM fHCM2__Team_Member__c WHERE fHCM2__Email__c = '%s' AND fHCM2__Has_Left__c = 'False'"
+
+type Record struct {
+	Employment_Record_Active__c bool
+}
+
+type Client struct {
+	salesforceClient SalesforceClientAPI
+
+	tracer  tracing.TracingInterface
+	monitor monitoring.MonitorInterface
+	logger  logging.LoggerInterface
+}
+
+func NewSalesforceClient(domain, consumerKey, consumerSecret string) (*salesforce.Salesforce, error) {
+	return salesforce.Init(salesforce.Creds{
+		Domain:         domain,
+		ConsumerKey:    consumerKey,
+		ConsumerSecret: consumerSecret,
+	})
+}
+
+func (c *Client) doAndMonitor(mail string) ([]Record, error) {
+	startTime := time.Now()
```

Comment:
let's add a tracing span here if we can

Reply:
Isn't the trace at https://github.com/canonical/user-verification-service/blob/main/pkg/user_verification/service.go#L21 enough?

Reply:
i missed that, it is indeed